roles/pbx/tasks/asterisk.yml

```diff
@@ -0,0 +1,159 @@
+  fail:
+    msg: "{ downloads_dir }}/{{ asterisk_src_file }} is REQUIRED in order to install."
+  when: not asterisk_src.stat.exists
+
+- name: Asterisk - Create install source directory
+  file: 
+    path: "{{ asterisk_src_dir }}"
+    state: directory
+
+- name: Asterisk - Extract source
+  unarchive: 
+    src: "{{ downloads_dir }}/{{ asterisk_src_file }}"
+    dest: "{{ asterisk_src_dir }}"
+    owner: root
+    group: root
+    extra_opts: [--strip-components=1]
+
+- name: Asterisk - Download mp3 decoder library into source tree
+  command: "./contrib/scripts/get_mp3_source.sh"
+  args:
+    chdir: "{{ asterisk_src_dir }}"
+    creates: "addons/mp3/mpg123.h"
+
+- name: Install aptitude (otherwise install_prereq fails)
+  package: 
+    name: aptitude 
+    state: latest
+
+- name: Asterisk - Ensure all dependencies are resolved
+  shell: export DEBIAN_FRONTEND=noninteractive && ./contrib/scripts/install_prereq install 
+  args:
+    chdir: "{{ asterisk_src_dir }}"
+
+- name: Asterisk - Run the configure script
+  command: "./configure"
+  args:
+    chdir: "{{ asterisk_src_dir }}"  
+
+- name: Asterisk - Run make menuselect.makeopts
+  command: "make menuselect.makeopts"
+  args:
+    chdir: "{{ asterisk_src_dir }}"
+
+- name: Asterisk - Do a bit of menuselect configuration
+  command: >
+    menuselect/menuselect --enable app_macro --enable format_mp3
+    --enable CORE-SOUNDS-EN-WAV --enable CORE-SOUNDS-EN-G722
+    --enable EXTRA-SOUNDS-EN-WAV --enable EXTRA-SOUNDS-EN-G722 --enable EXTRA-SOUNDS-EN-GSM
+    --disable-category MENUSELECT_MOH        
+  args:
+    chdir: "{{ asterisk_src_dir }}"
+
+- name: Asterisk - Run 'make'
+  command: make 
+  args:
+    chdir: "{{ asterisk_src_dir }}"
+
+- name: Asterisk - install
+  command: make install
+  args:
+    chdir: "{{ asterisk_src_dir }}"
+
+- name: Asterisk - install config
+  command: make config
+  args:
+    chdir: "{{ asterisk_src_dir }}"
+
+- name: Asterisk - install samples
+  command: make samples
+  args:
+    chdir: "{{ asterisk_src_dir }}"
+
+- name: Asterisk - ldconfig
+  command: ldconfig
+  args:
+    chdir: "{{ asterisk_src_dir }}"
+
+- name: Asterisk - Ensure group "asterisk" exists
+  group:
+    name: asterisk
+    state: present
+
+- name: Asterisk - Ensure user "asterisk" exists, and belongs to the required groups
+  user:
+    name: asterisk
+    group: asterisk
+    groups: audio,dialout
+    home: "/var/lib/asterisk"
+    system: yes
+    append: yes
+
+- name: Asterisk - Set directory ownership
+  file: 
+    dest: "{{ item }}" 
+    owner: asterisk 
+    group: asterisk
+    recurse: yes
+  with_items:
+    - /etc/asterisk
+    - /var/lib/asterisk
+    - /var/log/asterisk
+    - /var/spool/asterisk
+    - /usr/lib/asterisk
+
+- name: Asterisk - Set default user to asterisk in /etc/default/asterisk
+  lineinfile:
+      path: /etc/default/asterisk
+      regexp: 'AST_USER='
+      line: 'AST_USER="asterisk"'
```

Comment:
```suggestion
    line: 'AST_USER="asterisk"'
```

Reply:
sorry, vim wants to do this by default, and it is painful to fix it everytime. musthave sneaked thru

Reply:
will fix all these in one go